Dockerfile.s390x

```diff
@@ -30,46 +53,100 @@ RUN git clone https://github.com/apache/arrow.git && \
           -DARROW_WITH_SNAPPY=ON \
           -DARROW_JSON=ON \
           -DARROW_CSV=ON \
+          -DARROW_DATASET=ON \
           -DPROTOBUF_PROTOC_EXECUTABLE=/usr/bin/protoc \
           -DARROW_DEPENDENCY_SOURCE=BUNDLED \
-          -DARROW_DATASET=libarrow_dataset.so \
           .. && \
-    make -j$(nproc) && make install
+    make -j$(nproc) && \
+    make install && \
+    cd ../../python && \
+    export PYARROW_PARALLEL=4 && \
+    export ARROW_BUILD_TYPE=release && \
+    python -m pip install -r requirements-build.txt && \
+    python setup.py build_ext --build-type=$ARROW_BUILD_TYPE --bundle-arrow-cpp bdist_wheel
+
 
-ENV CMAKE_PREFIX_PATH=/usr/local/lib/cmake:$CMAKE_PREFIX_PATH
-ENV LD_LIBRARY_PATH=/usr/local/lib:$LD_LIBRARY_PATH
+# Install numactl (needed for numa.h dependency)
+WORKDIR /tmp
+RUN curl -LO https://github.com/numactl/numactl/archive/refs/tags/v2.0.16.tar.gz && \
+    tar -xvzf v2.0.16.tar.gz && \
+    cd numactl-2.0.16 && \
+    ./autogen.sh && \
+    ./configure && \
+    make && \
+    make install
 
+# Clean up build files
+RUN rm -rf /tmp/numactl-2.0.16 /tmp/v2.0.16.tar.gz
+
+# Set include path
+ENV C_INCLUDE_PATH="/usr/local/include:$C_INCLUDE_PATH"
+
+# Copy vLLM source
 COPY . /workspace/vllm
 WORKDIR /workspace/vllm
 
+# Check git repository integrity if enabled
 ARG GIT_REPO_CHECK=0
 RUN --mount=type=bind,source=.git,target=.git \
     if [ "$GIT_REPO_CHECK" != 0 ]; then bash tools/check_repo.sh; fi
 
-RUN mkdir -p /root/.cache/pip && chmod -R 777 /root/.cache/pip
-
+# Install dependencies, including PyTorch and Apache Arrow
 RUN --mount=type=cache,target=/root/.cache/pip \
     --mount=type=cache,target=/root/.cache/ccache \
     pip install --no-cache-dir -v \
         'cmake>=3.26' ninja packaging 'setuptools-scm>=8' wheel jinja2 \
+         /tmp/arrow/python/dist/*.whl  \
         --extra-index-url https://download.pytorch.org/whl/nightly/cpu \
         -r requirements-cpu.txt
 
+#Clean up build files for arrow
+RUN rm -rf /tmp/arrow
+
+# Install torchvision
 ARG TORCH_VISION_VERSION=v0.20.1
+WORKDIR /tmp
 RUN git clone https://github.com/pytorch/vision.git && \
     cd vision && \
     git checkout $TORCH_VISION_VERSION && \
-    python3 setup.py bdist_wheel && \
-    pip install --no-cache-dir dist/*.whl && \
-    rm -rf dist
+    python setup.py bdist_wheel && \
+    pip install --no-cache-dir dist/*.whl
```

Comment:
Why are we building this from scratch? This is part of `requirements-cuda.txt` and can be installed from PyPi or the torch wheel index